haas/ext/switches/brocade.py

```diff
@@ -0,0 +1,165 @@
+# Copyright 2016 Massachusetts Open Cloud Contributors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the
+# License. You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS
+# IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language
+# governing permissions and limitations under the License.
+
+"""A switch driver for the Brocade SDN
+
+Currently the driver uses telnet to connect to the switch's console; in the
+long term we want to be using SNMP.
+"""
+
+import json
+import logging
+import pexpect
+import re
+import requests
+import schema
+
+from sqlalchemy import Column, String, Integer, ForeignKey
+
+from haas.model import Switch
+
+logger = logging.getLogger(__name__)
+
+
+class Brocade(Switch):
+    api_name = 'http://schema.massopencloud.org/haas/v0/switches/brocade'
+
+    __mapper_args__ = {
+        'polymorphic_identity': api_name,
+    }
+
+    id = Column(Integer, ForeignKey('switch.id'), primary_key=True)
+    hostname = Column(String, nullable=False) # Hostname of the SDN
+    username = Column(String, nullable=False) # Username of the SDN
+    password = Column(String, nullable=False) # Password of the SDN
+    node = Column(String, nullable=False) # Name as registered in SDN
+    interface_type = Column(String, nullable=False)
+    dummy_vlan = Column(String, nullable=False)
```

Comment:
Are the fields `interface_type` and `dummy_vlan` required while registering a switch ?